#!/usr/bin/env Rscript
library(rmarkdown)

# Work around  https://github.com/rstudio/rmarkdown/issues/1508
# If the symbolic link is not replaced by a physical file
# output- and temporary files will be written to the original directory.
file.copy("./report_template.Rmd", "./template.Rmd", overwrite = TRUE)

rmarkdown::render("template.Rmd", output_file = "summary_report.html", params = list(css='nf-core_style.css',report_logo='nf-core-ampliseq_logo_light_long.png',workflow_manifest_version='2.15.0',workflow_scriptid='bf04a499cc',report_title='Summary of analysis results',metadata='Metadata.tsv',input_samplesheet='Samplesheet.tsv',mqc_plot='multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg',cutadapt_summary='cutadapt_summary.tsv',truncq=2,trunc_qmin=25,trunc_rmin=0.75,trunclenf='null',trunclenr='null',max_ee=2,dada_qc_f_path='FW_qual_stats.svg',dada_qc_r_path='RV_qual_stats.svg',dada_pp_qc_f_path='FW_preprocessed_qual_stats.svg',dada_pp_qc_r_path='RV_preprocessed_qual_stats.svg',dada_filtntrim_args='filterAndTrim.args.txt',dada_sample_inference='independent',dada_err_path='1_1.err.svg,1_2.err.svg',dada_err_run='1',asv_table_path='ASV_table.tsv',path_asv_fa='ASV_post_clustering_filtered.fna',path_dada2_tab='DADA2_table.tsv',dada_stats_path='DADA2_stats.tsv',mergepairs_strategy='merge',vsearch_cluster='ASV_post_clustering_filtered.table.tsv',vsearch_cluster_id='0.97',path_barrnap_sum='summary.tsv',filter_ssu_stats='stats.ssu.tsv',filter_ssu_asv='ASV_seqs.ssu.fasta',filter_ssu='bac',filter_len_asv='stats.len.tsv',filter_len_asv_len_orig='ASV_len_orig.tsv',min_len_asv=0,max_len_asv=255,dada_min_boot=50,dada2_taxonomy='ASV_tax_species.gtdb_R07-RS207.tsv',dada2_ref_tax_title='GTDB - Genome Taxonomy Database - Release R07-RS207',dada2_ref_tax_file='[https://data.gtdb.ecogenomic.org/releases/release207/207.0/genomic_files_reps/bac120_ssu_reps_r207.tar.gz, https://data.gtdb.ecogenomic.org/releases/release207/207.0/genomic_files_reps/ar53_ssu_reps_r207.tar.gz]',dada2_ref_tax_citation='Parks DH, Chuvochina M, Waite DW, Rinke C, Skarshewski A, Chaumeil PA, Hugenholtz P. A standardized bacterial taxonomy based on genome phylogeny substantially revises the tree of life. Nat Biotechnol. 2018 Nov;36(10):996-1004. doi: 10.1038/nbt.4229. Epub 2018 Aug 27. PMID: 30148503.',cut_dada_ref_taxonomy='assignTaxonomy.cutadapt.log',qiime2_taxonomy='taxonomy.tsv',qiime2_ref_tax_title='Greengenes 16S - Version 13_8 - clustered at 85% similarity - for testing purposes only',qiime2_ref_tax_file='[https://data.qiime2.org/2023.7/tutorials/training-feature-classifiers/85_otus.fasta, https://data.qiime2.org/2023.7/tutorials/training-feature-classifiers/85_otu_taxonomy.txt]',qiime2_ref_tax_citation='McDonald, D., Price, M., Goodrich, J. et al. An improved Greengenes taxonomy with explicit ranks for ecological and evolutionary analyses of bacteria and archaea. ISME J 6, 610â€“618 (2012). https://doi.org/10.1038/ismej.2011.139',val_used_taxonomy='DADA2',filter_stats_tsv='count_table_filter_stats.tsv',qiime2_filtertaxa='245,42',exclude_taxa='mitochondria,chloroplast',min_frequency='10',min_samples='2',barplot=TRUE,metadata_category_barplot='treatment1,badpairwise10',abundance_tables=TRUE,alpha_rarefaction=TRUE,diversity_indices_depth='WARNING\ The\ sampling\ depth\ of\ 543\ seems\ too\ small\ for\ rarefaction.txt',diversity_indices_alpha=TRUE,diversity_indices_beta='beta_diversity/weighted_unifrac_distance_matrix-mix8,beta_diversity/bray_curtis_distance_matrix-treatment1,beta_diversity/jaccard_distance_matrix-mix8,beta_diversity/unweighted_unifrac_distance_matrix-treatment1,beta_diversity/jaccard_distance_matrix-treatment1,beta_diversity/unweighted_unifrac_distance_matrix-mix8,beta_diversity/bray_curtis_distance_matrix-mix8,beta_diversity/weighted_unifrac_distance_matrix-treatment1',diversity_indices_adonis='beta_diversity/adonis/bray_curtis_distance_matrix-treatment1,beta_diversity/adonis/weighted_unifrac_distance_matrix-treatment1,beta_diversity/adonis/unweighted_unifrac_distance_matrix-treatment1,beta_diversity/adonis/jaccard_distance_matrix-mix8,beta_diversity/adonis/bray_curtis_distance_matrix-mix8,beta_diversity/adonis/weighted_unifrac_distance_matrix-mix8,beta_diversity/adonis/unweighted_unifrac_distance_matrix-mix8,beta_diversity/adonis/jaccard_distance_matrix-treatment1',qiime_adonis_formula='treatment1,mix8',ancombc='ancombc/da_barplot/Category-treatment1-ASV,ancombc/da_barplot/Category-treatment1-level-3,ancombc/da_barplot/Category-treatment1-level-2,ancombc/da_barplot/Category-treatment1-level-4,ancombc/da_barplot/Category-mix8-ASV,ancombc/da_barplot/Category-mix8-level-2,ancombc/da_barplot/Category-mix8-level-4,ancombc/da_barplot/Category-mix8-level-3,ancombc/da_barplot/Category-badpairwise10-level-4,ancombc/da_barplot/Category-badpairwise10-ASV,ancombc/da_barplot/Category-badpairwise10-level-2,ancombc/da_barplot/Category-badpairwise10-level-3',ancombc_formula='ancombc_formula/da_barplot/Category-treatment1-level-4,ancombc_formula/da_barplot/Category-treatment1-ASV,ancombc_formula/da_barplot/Category-treatment1-level-2,ancombc_formula/da_barplot/Category-treatment1-level-3',sbdi='sbdi/asv-table.tsv,sbdi/dna.tsv,sbdi/emof.tsv,sbdi/event.tsv,sbdi/annotation.tsv',phyloseq='phyloseq/dada2_phyloseq.rds,phyloseq/qiime2_phyloseq.rds',tse='tse/dada2_TreeSummarizedExperiment.rds,tse/qiime2_TreeSummarizedExperiment.rds'), envir = new.env())

writeLines(c("\"NFCORE_AMPLISEQ:AMPLISEQ:SUMMARY_REPORT\":",
    paste0("    R: ", paste0(R.Version()[c("major","minor")], collapse = ".")),
    paste0("    rmarkdown: ", packageVersion("rmarkdown")),
    paste0("    knitr: ", packageVersion("knitr")) ),
    "versions.yml")
